setup.py

```diff
@@ -73,7 +73,7 @@ def read(path):
               'createcoverage>=1,<2',
               'stopit>=1.1.2,<2',
               'flake8>=4,<5'],
-        doc=['sphinx>=3,<4',
+        doc=['sphinx>=3.5,<6',
```

Comment:
Python 3.10 support was added in Sphinx 4.3.0, so Sphinx in this project didn't run for me. This updates the version constraint to be aligned with the latest crate-docs-theme:

https://github.com/crate/crate-docs-theme/blob/66b054f08d0eab4c809a86a420c707c56dd7271e/setup.py#L61

Reply:
I am fine with relaxing the upper bound. What about this change?
```suggestion
        doc=['sphinx>=3,<6',
```

Reply:
Oh, same as `crate-docs-theme`. All right!